mesonbuild/dependencies/base.py

```diff
@@ -1901,6 +1879,8 @@ def get_methods():
 
 class ExternalProgram:
     windows_exts = ('exe', 'msc', 'com', 'bat', 'cmd')
+    # An 'ExternalProgram' always runs on the build machine
+    for_machine = MachineChoice.BUILD
```

Comment:
this doesn't have to be true with an exe_wrapper. I don't think that currently this changes behavior, but it might be valuable to be able to run a tool targeting the host machine.

Reply:
Oh good catch. I'm not at all sure looking at the code now. @jpakkane?

Reply:
We can add support for that later, right? If yes then this should work for now.